### PR TITLE
Adding was/configurations APIs to V3 package

### PR DESCRIPTION
**Pull request by ngaglani-tenable**
* Adding WAS Configurations API to V3 package.

### CODE REVIEW

File: tenable/io/v3/was/configurations/api.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+    :members:
+'''
+from typing import Dict, List, Optional
+from uuid import UUID
+
+from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
+from tenable.io.v3.was.configurations.schema import ConfigurationSchema
+
+
+class ConfigurationsAPI(ExploreBaseEndpoint):
```
Comment (tomeara) [inline tenable/io/v3/was/configurations/api.py:21]: Can we make a generic base class for both WAS and VM here?

Comment (ngaglani-tenable) [reply tenable/io/v3/was/configurations/api.py:21]: @tomeara, We are using `ExploreBaseEndpoint` as the generic base class for all the APIs, i.e., whether they are under `vm` or `was` or on the outer level. The configurations API is present only in WAS, hence, we have kept it under `was` only.
Please let us know if we are missing anything over here or if there are any further changes required in the class design/ structure.

**Review by sahils-tenable — APPROVED** [review]

File: tenable/io/v3/was/__init__.py [hunk 2]
```diff
@@ -40,6 +42,14 @@ def attachments(self):
         '''
         return AttachmentsAPI(self._api)
 
+    @property
+    def configurations(self):
+        '''
+        The interface object for the
+        :doc:`Configurations API <was-v2-configurations>`
```
[the remark this answers is not in the document]
Comment (ngaglani-tenable) [reply tenable/io/v3/was/__init__.py]: Done.

File: tenable/io/v3/was/configurations/api.py [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+            ...     'Destination Folder'
+            ... )
+
+        '''
+        payload = self._schema.dump(self._schema.load(
+            {
+                'folder_name': folder_name
+            }
+        ))
+        return self._patch(str(config_id), json=payload)
```
[the remark this answers is not in the document]
Comment (ngaglani-tenable) [reply tenable/io/v3/was/configurations/api.py]: Done.

File: tenable/io/v3/was/configurations/api.py [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+            :obj:`dict`:
+                The details of the specified scan configuration.
+
+        Examples:
+            >>> tio.v3.was.configurations.details(
+            ...     '00000000-0000-0000-0000-000000000000'
+            ... )
+        '''
+
+        return super().details(config_id)
```
Comment (parvdesai-tenable) [inline tenable/io/v3/was/configurations/api.py]: super()._details()

**Review by parvdesai-tenable — APPROVED** [review]

File: tenable/io/v3/was/configurations/api.py (hunk 4 above)
Comment (ngaglani-tenable) [reply tenable/io/v3/was/configurations/api.py]: Done

File: tenable/io/v3/was/configurations/api.py [hunk 5]
```diff
@@ -0,0 +1,409 @@
+'''
+Configurations
+==============
+
+The following methods allow for interaction into the Tenable.io
```
Comment (Mangusingh-te) [inline tenable/io/v3/was/configurations/api.py:5]:
```suggestion
The following methods allow for interaction with Tenable.io
```

File: tenable/io/v3/was/configurations/api.py [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+                scan.
+            settings (Dict):
+                The scan configuration settings.
+            description (str):
+                A description of the scan configuration.
+            folder_id (str):
+                The UUID of the folder to assign for the scan configuration.
+            user_template_id (str):
+                The UUID of the user-defined configuration template from which
+                this configuration was derived.
```
Comment (Mangusingh-te) [inline tenable/io/v3/was/configurations/api.py:66]:
```suggestion
                this configuration is derived.
```

File: tenable/io/v3/was/configurations/api.py [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+            limit (int, optional):
+                Number of objects to be returned in each request.
+                Default and maximum limit is 200.
+            offset (int, optional):
+                The pagination offset to use when requesting the next page of
+                results.
+            num_pages (int, optional):
+                The total number of pages to request before stopping the
+                iterator.
+            return_resp (bool, optional):
```
Comment (Mangusingh-te) [inline tenable/io/v3/was/configurations/api.py:289]: Update docstring.

File: tenable/io/v3/was/configurations/api.py [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+            return_csv (bool, optional):
+                If set to true, it will return the CSV response or
+                iterable (based on return_resp flag). Iterator returns all
+                rows in text/csv format for each call with row headers.
+
+        Returns:
+            Iterable:
+                The iterable that handles the pagination for the job.
+            requests.Response:
+                If ``return_json`` was set to ``True``, then a response
```
Comment (Mangusingh-te) [inline tenable/io/v3/was/configurations/api.py:302]: Update docstring.

File: tenable/io/v3/was/configurations/schema.py [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    folder_name = fields.String()
+    name = fields.String()
+    notifications = fields.Dict(keys=fields.String(validate=v.Equal('emails')),
+                                values=fields.List(fields.Email()))
+    owner_id = fields.UUID()
+    permissions = fields.List(fields.Nested(PermissionConfigurationSchema))
+    scanner_id = fields.UUID()
+    schedule = fields.Nested(ScheduleSchema)
+    settings = fields.Nested(SettingsSchema)
+    targets = fields.List(fields.Url())
```
Comment (Mangusingh-te) [inline tenable/io/v3/was/configurations/schema.py:84]: Can targets accept list of ip-addresses of webapps to be scanned ? If yes, do **fields.url()** support that ?

File: tests/io/v3/was/configurations/test_configurations_api.py [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+
+    iterator = api.v3.was.configurations.search(
+        fields=fields, limit=200, sort=sort, filter=filters
+    )
+    assert isinstance(iterator, SearchIterator)
+
+    configurations_list = []
+    for configuration in iterator:
+        configurations_list.append(configuration)
+    assert len(configurations_list) == api_response['pagination']['total']
```
Comment (Mangusingh-te) [inline tests/io/v3/was/configurations/test_configurations_api.py:307]: get rid of loop.

**Review by Mangusingh-te — CHANGES_REQUESTED** [review]

**Comment by MitaliUpadhyay** [comment]
@alpesh-te  , please close this PR as changes are already addressed in https://github.com/alpesh-te/pyTenable/pull/2

**Pull request closed without merging**